app-misc/pip3line/pip3line-3.4.2.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6,7}} )
+
+inherit cmake-utils python-r1
+
+DESCRIPTION="Raw bytes manipulation, transformations (decoding and more) and interception"
+HOMEPAGE="https://github.com/metrodango/pip3line"
+
+KEYWORDS=""
+SRC_URI=""
+
+if [[ ${PV} == 9999* ]] ; then
+    inherit git-r3
+    EGIT_REPO_URI="https://github.com/metrodango/pip3line.git"
+    EGIT_BRANCH="master"
+else
+    SRC_URI="https://github.com/metrodango/pip3line/archive/v${PV}.tar.gz  -> ${P}.tar.gz"
+    KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+
+IUSE="distorm openssl qscintilla"
+
+RDEPEND="
+    ${PYTHON_DEPS}
+    dev-qt/qtconcurrent:5
+    dev-qt/qtcore:5
+    dev-qt/qtgui:5
+    dev-qt/qtnetwork:5
+    dev-qt/qtsvg:5
+    dev-qt/qtwidgets:5
+    dev-qt/qtxmlpatterns:5
+    distorm? ( dev-libs/distorm64 )
+    openssl? ( dev-libs/openssl:0= )
+    qscintilla? ( x11-libs/qscintilla )"
+
+DEPEND="${RDEPEND}"
+
+src_configure() {
+
+    if [[ "${PYTHON_TARGETS}" == *python3* ]]; then
+        python3_usage=yes
+    else
+        python3_usage=no
+    fi
+
+    local mycmakeargs=(
+        -DBASIC=yes
+        -DWITH_OPENSSL=$(usex openssl)
+        -DWITH_PYTHON27=$(usex python_targets_python2_7)
+        -DWITH_PYTHON3=${python3_usage}
+        -DWITH_SCINTILLA=$(usex qscintilla)
+        -DWITH_DISTORM=$(usex distorm)
+    )
+
+    cd ext || die
```

Comment:
Shouldn't this be use-conditional? Also, this requires git, which isn't documented as a dependency. Also, you link to a git repo and then run with system-installed `distorm64`? Wouldn't this lead to problems in case of ABI mismatch? Shouldn't you build against system-installed distorm?

Reply:
I was told to remove git as a dependency in a previous PR, you people need to decide which is it.

For distomr64 please see comment below.

Reply:
The previous commenter probably thought that your git dependency came from the `9999` ebuild, and for that he was right: `inherit git-r3` takes care of that. But in this case, `gen-distorm.sh` needs git and `git-r3` isn't inherited in non-live ebuilds, so a git dependency is needed.

As for distorm, if the system package isn't usable, then I suggest that you drop the dependency entirely and link statically, with a comment explaining why. This current situation invites troubles.

Reply:
There was not much of an explanation in the previous PR, but it now makes sense.

This change for distorm is going to require a new release .... so I am guessing I need to close this PR, and go through the cycle again.

Reply:
No need to close the PR. Of course, since you're upstream, you control the release cycle, but you could also add patches to your ebuild to force static linking. No need to release a v3.4.3.

Reply:
I have now integrated this step into the cmake script, with a cmake option to compile the static distorm lib when needed. it's much cleaner this way.

I was about to release a new upstream version anyway.